instance/main.tf

```diff
@@ -1,6 +1,19 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 3.0"
+    }
+  }
+}
+
+provider "aws" {
+  region = "us-east-1"
+}
+
 resource "aws_security_group" "web-sg" {
-  name   = "${var.env_code}-sg"
-  vpc_id = aws_vpc.main.id
+  name   = "${data.terraform_remote_state.networking.outputs.env_code}-sg"
```

Comment:
Please don't use a remote state for this variable. Instead, declare it as a separate variable in level2.
The name can remain the same: level2